### PR TITLE
feat: add support for Ruby v3

### PR DESCRIPTION
**Pull request by DevinCodes**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no (?)
| Need Doc update   | yes/no

## Describe your change
This PR is meant to add support for version 3 of Ruby, which was released on 25 December 2020.

The following dependencies don't support Ruby v3 yet, meaning we can't upgrade for now:
- [`net-http-persistent`](https://github.com/drbrain/net-http-persistent/) (https://github.com/drbrain/net-http-persistent/issues/123)

### CODE REVIEW

**Comment by mcansky** [comment]
this would be great, we are blocked by the missing support to ruby 3.0 ; we are looking into writing a minimal REST API client to cover our limited needs in the meantime

**Comment by DevinCodes** [comment]
Seems like support has been added in the meantime (see https://github.com/drbrain/net-http-persistent/commit/b07434826c784ee514a28824f7dc4810b7a2ba35). I'll try to take some time this week to make the tests here green 🙂

**Comment by acallaghan** [comment]
Any news on this?

**Comment by DevinCodes** [comment]
This should already work without this PR being merged, feel free to test it out and let me know if this is indeed the case. We're having issues with the CI that keeps failing for now, so we'll have to dig deeper into why this is happening.

**Comment by Keqi** [comment]
Hi @DevinCodes,

Did you have a chance to fix your CI issues? I would love to use official Ruby 3 support in your gem :)

**Comment by DevinCodes** [comment]
@Keqi I'll be releasing Ruby v3 support next week 🙂

**Comment by Keqi** [comment]
Perfect, I'll watch this PR patiently then :)

File: test/algolia/integration/search_client_test.rb [hunk 1]
```diff
@@ -409,14 +409,21 @@ def test_stopwords_dictionaries
         stopwords_settings = {
           disableStandardEntries: {
             stopwords: {
-              en: true
+              en: true,
+              fr: false
```
Comment (DevinCodes) [inline test/algolia/integration/search_client_test.rb:413]:
```suggestion
              en: true
```

File: test/algolia/integration/search_client_test.rb [hunk 2]
```diff
@@ -409,14 +409,21 @@ def test_stopwords_dictionaries
...
         @client.set_dictionary_settings!(stopwords_settings)
 
-        assert_equal @client.get_dictionary_settings, stopwords_settings
+        assert_equal @client.get_dictionary_settings, {
+          disableStandardEntries: {
+            stopwords: {
+              en: true
+            }
+          }
+        }
```
Comment (DevinCodes) [inline test/algolia/integration/search_client_test.rb:426]:
```suggestion
        assert_equal @client.get_dictionary_settings, stopwords_settings
```

**Review by TomKlotzPro — APPROVED** [review]
LGTM! Let's merge this 🚀  Thank you for this work @DevinCodes !

**Pull request merged**